odoo/http.py

```diff
@@ -1142,6 +1142,77 @@ def update_trace(self, request):
     def _delete_old_sessions(self):
         root.session_store.delete_old_sessions(self)
 
+    def _update_session_token(self, env: odoo.api.Environment | None = None):
+        """
+        Compute the session token of the current user (determined using
+        ``session['uid']``) and save it in the session.
+
+        :param env: An environment to access ``res.users``. It uses the
+            environment attached to the current request by default.
+        """
+        if env is None:
+            env = request.env
+        user = env['res.users'].browse(self['uid'])
+        self['session_token'] = user._compute_session_token(self.sid)
+
+    def _check(
+        self,
+        *,
+        request: 'Request | None' = None,
```

Comment:
Make it mandatory: it makes the change easier to understand instead of magically using a global variable.

`from future import __annotations__` instead of string types.